addons/mail/static/src/js/many2one_avatar_user.js

```diff
@@ -0,0 +1,136 @@
+odoo.define('mail.Many2OneAvatarUser', function (require) {
+    "use strict";
+
+    // This module defines an extension of the Many2OneAvatar widget, which is
+    // integrated with the messaging system. The Many2OneAvatarUser is designed
+    // to display people, and when the avatar of those people is clicked, it
+    // opens a DM chat window with the corresponding partner (the messaging
+    // system is based on model 'res.partner').
+    //
+    // This widget is supported on many2one fields pointing to 'res.users'. When
+    // the user is clicked, we fetch the partner id of the given user (field
+    // 'partner_id'), and we open a chat window with this partner.
+    //
+    // Usage:
+    //   <field name="user_id" widget="many2one_avatar_user"/>
+    //
+    // The widget is designed to be extended, to support many2one fields pointing
+    // to other models than 'res.users'. Those models must have a many2one field
+    // pointing to 'res.partner'. Ideally, the many2one should be false if the
+    // corresponding partner isn't associated with a user, otherwise it will open
+    // chat windows with partners that won't be able to read your messages and
+    // reply.
+
+    const { _t } = require('web.core');
+    const fieldRegistry = require('web.field_registry');
+    const { Many2OneAvatar } = require('web.relational_fields');
+    const session = require('web.session');
+
+
+    const Many2OneAvatarUser = Many2OneAvatar.extend({
+        events: Object.assign({}, Many2OneAvatar.prototype.events, {
+            'click .o_m2o_avatar': '_onAvatarClicked',
+        }),
+        // Maps record ids to promises that resolve with the corresponding partner ids.
+        // Used as a cache shared between all instances of this widget.
+        partnerIds: {},
+        // This widget is only supported on many2ones pointing to 'res.users'
+        supportedModel: 'res.users',
+
+        init() {
+            this._super(...arguments);
```

Comment:
Actually, the Many2OneAvatar template is really simple: an img and a span. So it won't display a link, never. I forced noOpen to true s.t. if the user clicks on the display_name (which doesn't look like something clickable), it doesn't open the record. A link next to the img looks weird. In our two usecases, we don't want the link to redirect to the record. So here I see 3 options:
  - you agree that this is fine and we keep it as it is
  - instead of forcing noOpen to true, I can remove the handlers (maybe more elegant)
  - I change Many2OneAvatar to behave more closely like the classical many2one (thus displaying a link if noOpen is not set to true)

Note that I tried solution 3, but if I remember correctly it wasn't that easy (because in this case I want to reuse the same template). Don't remember the details though. We can talk about this.